src/adapters/kafka/index.ts

```diff
@@ -13,47 +12,24 @@ class KafkaAdapter extends Adapter {
 
   async connect() {
     const brokerUrl = this.AsyncAPIServer.url()
-    // creating client with broker Url & initiating Kafkajs client by pointing it towards at least one broker
-    this.kafka = new Kafka({
+    const host = process.env.HOST_IP || ip.address()
+   
+    const kafka = new Kafka({
       clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
-      brokers: [brokerUrl],
-      ssl: {
-        rejectUnauthorized: false,
-        ca: [fs.readFileSync('/my/custom/ca.crt', 'utf-8')],
-        key: fs.readFileSync('/my/custom/client-key.pem', 'utf-8'),
-        cert: fs.readFileSync('/my/custom/client-cert.pem', 'utf-8')
-      },
-    })
-
-    // creating a producer using the client to produce a message to a topic 
-    // Kafka requires that the transactional producer should have the following configuration to guarantee EoS-exactly once semantics, 
-    // Configure the producer client with maxInFlightRequests: 1, idempotent: true and a transactionalId to guarantee EOS
-
-    const producer = this.kafka.producer({
-      transactionalId: 'my-transactional-producer',
-      maxInFlightRequests: 1,
-      idempotent: true
+      brokers: [`${host}:9092`, `${host}:9097`, `${host}:9100`],
     })
 
+    const producer = kafka.producer()
     await producer.connect()
     await producer.send({
       topic: 'test-topic',
       messages: [
         { value: 'Hello KafkaJS user!' },
       ],
     })
+    await producer.disconnect()
```

Comment:
You should not have a producer here. Otherwise, it will send a message when you connect to the broker.